src/pages/SignIn.jsx

```diff
@@ -38,19 +37,18 @@ const SignIn = () => {
           id="email"
           data-testid="email-input"
           value={emailInput.value}
-          placeholder="@를 포함하여 이메일을 입력해 주세요"
           onChange={e => emailInput.handleChange(e.target.value)}
         />
+        <ValidMessage $isValid={!emailInput.isValid}>@를 포함하여 이메일을 입력해 주세요</ValidMessage>
```

Comment:
```suggestion
        {emailInput.isValid && <ValidMessage>@를 포함하여 이메일을 입력해 주세요</ValidMessage>}
```
isValid를 이용해 처리하기 보단 emailInput.isValid로 컨트롤하는게 좋을 것 같습니다. visibility 속성은 UI가 사라지는게 아닌 숨겨졌다에 가깝습니다. 실제로 hidden 값을 가져도 HTML에는 엘리먼트가 남아있습니다. 반면 조건부 렌더링을 하게되면 HTML에도 사라집니다.

Reply:
사라질 때 컴포넌트 크기를 유지하고 싶어서 visibility로 적용했는데 
자리 차지하기 위해 display none, min-height를 설정하는게 나을까요?
아니면 컴포넌트 크기 상관없이 조건부렌더링으로 처리해서 html에서 아예 없애는게 좋을까요?

Reply:
아 영역을 지키고 싶으셔서 이렇게 하신거군요 그렇다면 visibility을 사용하신게 이해가 되네요! Input 컴포넌트의 min-height을 넣은 것도 같은 이유로 알고 있겠습니다. 감사합니다!